xla/service/gpu/nccl_collective_broadcast_thunk.cc

```diff
@@ -0,0 +1,125 @@
+#include "xla/service/gpu/ir_emission_utils.h"
+#include "xla/translate/mhlo_to_hlo/attribute_exporter.h"
+#include "xla/xla_data.pb.h"
+#include "xla/service/gpu/nccl_api.h"
+
+
+#if XLA_ENABLE_XCCL
+#include "xla/stream_executor/gpu/gpu_stream.h"
+#endif
+
+namespace xla {
+namespace gpu {
+
+using mlir::lmhlo_gpu::CollectiveBroadcastStartOp;
+
+namespace impl {
+
+CollectiveOpGroupMode GetGroupMode(CollectiveBroadcastStartOp op) {
+  return GetCollectiveOpGroupMode(op.getChannelId().has_value(), std::nullopt)
+      .value();
+}
+
+}  // namespace impl
+
+
+NcclCollectiveBroadcastStartThunk::NcclCollectiveBroadcastStartThunk(
+    ThunkInfo thunk_info, NcclApi* nccl_api, CollectiveBroadcastStartOp op, std::vector<Buffer> buffers)
+    : NcclCollectiveThunk(Thunk::kNcclCollectiveBroadcastStart, thunk_info, nccl_api,
+                          op.getIsSync()),
+      config_(GetNcclCollectiveConfigForMlir(op, std::nullopt)),
+      buffers_(std::move(buffers)) {}
+
+NcclCollectiveBroadcastStartThunk::NcclCollectiveBroadcastStartThunk(
+    ThunkInfo thunk_info, NcclApi* nccl_api,  const HloCollectiveBroadcastInstruction* instr, std::vector<Buffer> buffers)
+    : NcclCollectiveThunk(Thunk::kNcclCollectiveBroadcastStart, thunk_info, nccl_api,
+                          IsSyncCollective(instr)),
+      config_(GetNcclCollectiveConfig(instr, std::nullopt)),
+      buffers_(std::move(buffers)) {}
+
+/*static*/ Status NcclCollectiveBroadcastStartThunk::CheckImplementable(
+    CollectiveBroadcastStartOp op, int64_t replica_count,
+    int64_t partition_count) {
+  return OkStatus();
+}
+
+/*static*/ Status NcclCollectiveBroadcastStartThunk::CheckImplementable(
+    const HloInstruction* instr, int64_t replica_count,
+    int64_t partition_count) {
+  return OkStatus();
+}
+
+/*static*/ bool NcclCollectiveBroadcastStartThunk::IsDegenerate(
+    CollectiveBroadcastStartOp op, int64_t replica_count,
+    int64_t partition_count) {
+  return false;
+}
+
+/*static*/ CollectiveOpGroupMode NcclCollectiveBroadcastStartThunk::GetGroupMode(
+    CollectiveBroadcastStartOp op) {
+  return impl::GetGroupMode(op);
+}
+
+
+/*static*/ CollectiveOpGroupMode NcclCollectiveBroadcastStartThunk::GetGroupMode(
+    const HloCollectiveBroadcastInstruction* inst) {
+  return GetNcclCollectiveConfig(inst, std::nullopt).group_mode;
+}
+
+Status NcclCollectiveBroadcastStartThunk::RunNcclCollective(
+    const ExecuteParams& params, se::Stream& stream, NcclApi::NcclCommHandle comm) {
+  TF_ASSIGN_OR_RETURN(
+      std::vector<DeviceBufferPair> device_buffers,
+      ConvertToDeviceBuffers(params, buffers_,
+                             config_.operand_element_type));
+  return ::xla::gpu::RunCollectiveBroadcast(device_buffers, stream, comm, nccl_api());
+}
+
+Status RunCollectiveBroadcast(std::vector<DeviceBufferPair>& buffers, se::Stream& stream,
+                            NcclApi::NcclCommHandle comm, NcclApi* nccl_api) {
+#if XLA_ENABLE_XCCL
```

Comment:
this is also not needed, everything in XLA should work these days without a need to hide behind XLA_ENABLE_XCCL

Reply:
Good to know, I will remove it